BLSQ/iaso
# WC2-252 enable export of workflows to another environment

Enable the export of a (entity) workflow to a JSON
And then import of this JSON into (a possibly different instance of iaso).
(Example from staging to prod)


Related JIRA tickets : WC2-252

## Self proofreading checklist


## Changes

Add endpoints for import export.
Add UUIDs on workflows, versions, changes, follow_ups in order to handle the case of "re-import" ..
Cleanup repeated code in tests
Add some code in test_utils.py

## How to test

After creating a workflow, got to `/api/workflows/export/{workflow_id}/` and get the JSON.
Then try importing it on `/api/workflows/import/` (no need for is as there are UUIDs in the JSON).
(You can try to modify some parts of the JSON and see if they are applied)
(See the tests for some example)

## Notes

This DOES not import and export the entities, entity types or forms used in the workflows, these have to be migrated independently.
